### PR TITLE
add a discrete clone method to scopes

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Adds a method to all scopes that produces a
discrete clone of the original scope.  The result is a shallow clone of the scope, thus retaining all
original values, but is made discrete by replacing the resource owner target with a specific value.

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* #1621

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

File: src/pkg/selectors/onedrive.go [hunk 1]
```diff
@@ -459,6 +459,29 @@ func (s OneDriveScope) setDefaults() {
...
+	return discreteCopy(s, user)
+}
+
+// ---------------------------------------------------------------------------
+// Backup Details Filtering
+// ---------------------------------------------------------------------------
+
+// Reduce filters the entries in a details struct to only those that match the
+// inclusions, filters, and exclusions in the selector.
+func (s oneDrive) Reduce(ctx context.Context, deets *details.Details) *details.Details {
```
Comment (ryanfkeepers) [inline src/pkg/selectors/onedrive.go:474]: Weird diff: this func wasn't changed.  What changed is that `matchesInfo()` was moved below it.

File: src/pkg/selectors/sharepoint.go [hunk 2]
```diff
@@ -407,6 +407,30 @@ func (s SharePointScope) setDefaults() {
...
+// site comparison with only the provided site.
+func (s SharePointScope) DiscreteCopy(site string) SharePointScope {
+	return discreteCopy(s, site)
+}
+
+// ---------------------------------------------------------------------------
+// Backup Details Filtering
+// ---------------------------------------------------------------------------
+
+// Reduce filters the entries in a details struct to only those that match the
```
Comment (ryanfkeepers) [inline src/pkg/selectors/sharepoint.go:420]: Weird diff: this func wasn't changed.  What changed is that `matchesInfo()` was moved below it.

File: src/pkg/selectors/scopes_test.go [hunk 3]
```diff
@@ -513,3 +513,20 @@ func (suite *SelectorScopesSuite) TestScopeConfig() {
...
+	)
+
+	for k, v := range orig {
+		if k != rootCatStub.String() {
+			assert.Equal(t, v.Target, clone[k].Target)
+		} else {
+			assert.Equal(t, AnyTgt, v.Target)
+			assert.Equal(t, "fnords", clone[k].Target)
+		}
+	}
```
Comment (ashmrtn) [inline src/pkg/selectors/scopes_test.go:531]: could you please also add an assertion to make sure the original has not changed as well. I know it seems obvious, but sometimes assumptions like these are what come back to get you in the end

**Review by ashmrtn — APPROVED** [review]

File: src/pkg/selectors/scopes_test.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/pkg/selectors/scopes_test.go:531]: The assertion on 528 is precisely that.

Comment (ashmrtn) [reply src/pkg/selectors/scopes_test.go:531]: 🤦‍♀️

**Pull request merged**